frontend/src/metabase/query_builder/selectors.js

```diff
@@ -315,14 +315,21 @@ export const getLastRunQuestion = createSelector(
     card && metadata && new Question(card, metadata, parameterValues),
 );
 
-export const getQuestion = createSelector(
-  [getMetadata, getCard, getParameterValues, getQueryBuilderMode],
-  (metadata, card, parameterValues, queryBuilderMode) => {
+export const getDirtyQuestion = createSelector(
+  [getMetadata, getCard, getParameterValues],
+  (metadata, card, parameterValues) => {
     if (!metadata || !card) {
       return;
     }
-    const question = new Question(card, metadata, parameterValues);
-
+    return new Question(card, metadata, parameterValues);
+  },
+);
+export const getQuestion = createSelector(
+  [getDirtyQuestion, getQueryBuilderMode],
+  (question, queryBuilderMode) => {
```

Comment:
nit: this is to distinguish that we're receiving a dirty question. Not a question that we can use instantly everywhere.

p.s. Please take a look at my other comment about the `getDirtyQuestion` name too.
```suggestion
  (dirtyQuestion, queryBuilderMode) => {
```